File: DSApython/Algorithm/Sorting/bubbleSort.py
def bubbleSort(arr):
    n = len(arr)
    for i in range(n-1):
        sorted = True
        for j in range(n-i-1):
            if arr[j]>arr[j+1]:
                arr[j],arr[j+1] = arr[j+1],arr[j]
                sorted = False
        if sorted:
            break

arr = [1,2,3,4,10,9,8,7,6,5]
bubbleSort(arr)
print(arr)

File: DSAcpp/Algorithm/Sortings/bubbleSort.cpp
#include <iostream>
#include <vector>
using namespace std;

void printArr(const vector<int> &arr){
    for(int i:arr) cout << i << ' ';
    cout << '\n';
}

void bubbleSort(vector<int> &arr){
    bool sorted = false;
    int n = arr.size();
    for(int i=0;i<n && !sorted;i++){
        sorted = true;
        for(int j=0;j<n-i-1;j++){
            if(arr[j]>arr[j+1]) {
                swap(arr[j],arr[j+1]);
                sorted = false;
            }
        }
    }
}

int main(){
    vector<int> arr = {1,2,3,4,10,9,8,7,6,5};
    bubbleSort(arr);
    printArr(arr);
}